generic_throttle/src/generic_throttle/generic_throttle.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python
+
+import rospy
+import rostopic
+from threading import Lock
+from threading import currentThread
+from functools import partial
+
+
+class GenericThrottle:
+    def __init__(self):
+        self.topic_dictionary = None
+        self.parameters = {'namespace': None,
+                           'topic_framerate': None,
+                           'latched': False,
+                           'lazy': False}
+
+        rospy.init_node('generic_throttle')
+        rospy.on_shutdown(self._shutdown)
+
+        # Read mandatory /generic_throttle/* parameters from server
+        parameter_list = ['namespace', 'topic_framerate']
+        for parameter_name in parameter_list:
+            parameter_string = '/generic_throttle/' + parameter_name
+            if rospy.has_param(parameter_string):
+                self.parameters[parameter_name] = \
+                    rospy.get_param(parameter_string)
+            else:
+                rospy.logerr('Parameter ' + parameter_string + ' not available')
+                exit(5)
+
+        # Read optional /generic_throttle/* parameters from server
+        parameter_list = ['latched', 'lazy']
+        for parameter_name in parameter_list:
+            parameter_string = '/generic_throttle/' + parameter_name
```

Comment:
do not assume absolute namespace....just look for ```parameter_name``` under the node's namespace, i.e. ```parameter_string = parameter_name``` (of course then ```parameter_string``` is obsolete!)